src/WalkingModule/src/Module.cpp

```diff
@@ -521,27 +638,63 @@ bool WalkingModule::updateModule()
         // check desired planner input
         yarp::sig::Vector* desiredUnicyclePosition = nullptr;
         desiredUnicyclePosition = m_desiredUnyciclePositionPort.read(false);
-        if(desiredUnicyclePosition != nullptr)
-            if(!setPlannerInput((*desiredUnicyclePosition)(0), (*desiredUnicyclePosition)(1)))
+        if(m_useStepAdaptation)
+        {
+            if(desiredUnicyclePosition != nullptr)
             {
-                yError() << "[WalkingModule::updateModule] Unable to set the planner input";
-                return false;
+                if(!setPlannerInput((*desiredUnicyclePosition)(0), 0.0))
+                {
+                    yError() << "[WalkingModule::updateModule] Unable to set the planner input";
+                    return false;
+                }
             }
 
+            if (m_mergePoints.front() == 11 && desiredUnicyclePosition == nullptr)
+            {
+                if(!setPlannerInput(m_desiredPosition(0) ,m_desiredPosition(1)))
+                {
+                    yError() << "[updateModule] Unable to recall the setplannerInput (when terminal (SetGoal) instead of JoyStick is used)";
+                    return false;
+                }
+            }
+        }
+        else
+        {
+            if(desiredUnicyclePosition != nullptr)
+            {
+                if(!setPlannerInput((*desiredUnicyclePosition)(0), (*desiredUnicyclePosition)(1)))
+                {
+                    yError() << "[WalkingModule::updateModule] Unable to set the planner input";
+                    return false;
+                }
+            }
+        }
+
         // if a new trajectory is required check if its the time to evaluate the new trajectory or
         // the time to attach new one
         if(m_newTrajectoryRequired)
         {
             // when we are near to the merge point the new trajectory is evaluated
-            if(m_newTrajectoryMergeCounter == 20)
+            if(m_newTrajectoryMergeCounter == 10)
             {
 
                 double initTimeTrajectory;
                 initTimeTrajectory = m_time + m_newTrajectoryMergeCounter * m_dT;
 
-                iDynTree::Transform measuredTransform = m_isLeftFixedFrame.front() ?
-                    m_rightTrajectory[m_newTrajectoryMergeCounter] :
-                    m_leftTrajectory[m_newTrajectoryMergeCounter];
+                m_startOfWalkingTime=initTimeTrajectory;
+
+                iDynTree::Transform TempRightFoot;
+                iDynTree::Transform TempLeftFoot;
+                iDynTree::Transform measuredTransform ;
+
+                if(!m_useStepAdaptation){
+                    measuredTransform = m_isLeftFixedFrame.front() ?
+                                        m_rightTrajectory[m_newTrajectoryMergeCounter] : m_leftTrajectory[m_newTrajectoryMergeCounter];
```

Comment:
Why there is this difference with respect to the nominal case?

Reply:
In a case that step adaptation is active we have the `m_currentFootRightTransform` instead of  `m_rightTrajectory[m_newTrajectoryMergeCounter]` since we re-plan the feet transforms when step adjustment is active.

Reply:
Hi. Is it clear or you suggest some changes?

Reply:
No sorry, now I got it :+1: